packages-exp/auth-exp/src/core/credentials/oauth.ts

```diff
@@ -0,0 +1,161 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as externs from '@firebase/auth-types-exp';
+import { querystring } from '@firebase/util';
+
+import {
+  signInWithIdp,
+  SignInWithIdpRequest
+} from '../../api/authentication/idp';
+import { Auth } from '../../model/auth';
+import { IdTokenResponse } from '../../model/id_token';
+import { AuthErrorCode } from '../errors';
+import { fail } from '../util/assert';
+import { AuthCredential } from './';
+
+const IDP_REQUEST_URI = 'http://localhost';
+
+export interface OAuthCredentialParams {
+  idToken?: string | null;
+  accessToken?: string | null;
+  oauthToken?: string;
+  secret?: string;
+  oauthTokenSecret?: string;
+  nonce?: string;
+  pendingToken?: string;
```

Comment:
can you add some comments around which token is used in which case? we hve at least 4 different types of tokens in these params

Reply:
Done